modules/react/menu/lib/MenuGroup.tsx

```diff
@@ -0,0 +1,121 @@
+import {OverflowTooltip} from '@workday/canvas-kit-react/tooltip';
+import {FlexProps, mergeStyles} from '@workday/canvas-kit-react/layout';
+
+export interface MenuGroupHeaderProps extends CSProps, FlexProps {
+  /**
+   * The label text of the {@link MenuGroup Menu.Group}.
+   */
+  children?: React.ReactNode;
+}
+
+export interface MenuGroupProps extends CSProps, FlexProps {
+  /**
+   * The children of the `Menu.Group`. Should be {@link MenuGroupHeading Menu.Group.Heading} or
+   * {@link MenuItem Menu.Item}. If the `title` prop is provided, the `Menu.Group.Heading` is not
+   * needed.
+   */
+  children?: React.ReactNode;
+  /**
+   * Optional `Menu.Group` title. If a `title` is provided, a `Menu.Group.Heading` will be inserted
+   * with this prop being the `children`. If you wish more control over the styling of the
+   * `Menu.Group` heading, use the `<Menu.Group.Heading>` component directly and you'll have access
+   * to styling:
+   *
+   * ```tsx
+   * // default
+   * <Menu.Group title="Group One">
+   *   <Menu.Item>Item One</Menu.Item>
+   * </Menu.Group>
+   *
+   * // direct access to the heading
+   * <Menu.Group>
+   *   <Menu.Group.Heading cs={...}>Group One</Menu.Group.Heading>
+   *   <Menu.Item>Item One</Menu.Item>
+   * </Menu.Group>
+   * ```
+   */
+  title?: string;
+}
+
+const useMenuGroupModel = createModelHook({
+  defaultConfig: {
+    id: undefined as string | undefined,
+  },
+  requiredConfig: {},
+})(config => {
+  const state = {
+    id: useUniqueId(config.id),
+  };
+  const events = {};
+
+  return {
+    state,
+    events,
+  };
+});
+
+export const menuGroupHeadingStencil = createStencil({
+  base: {
+    ...system.type.subtext.large,
+    display: 'flex',
+    alignItems: 'center',
+    width: '100%',
+    gap: system.space.x4,
+    padding: `${system.space.x2} ${system.space.x4}`,
+    boxSizing: 'border-box',
+    fontWeight: system.fontWeight.bold,
+  },
+});
+
+export const useMenuGroup = createElemPropsHook(useMenuGroupModel)(model => {
+  return {
+    'data-type': 'group',
+    role: 'group',
+    'aria-labelledby': model.state.id,
+  };
+});
+
+export const useMenuGroupHeading = createElemPropsHook(useMenuGroupModel)(model => {
+  return {
+    id: model.state.id,
+  };
+});
+
+const MenuGroupHeading = createSubcomponent('div')({
+  modelHook: useMenuGroupModel,
+  elemPropsHook: useMenuGroupHeading,
+})(({children, ...elemProps}, Element) => {
+  return (
+    <OverflowTooltip>
+      <Element {...mergeStyles(elemProps, menuGroupHeadingStencil())}>{children}</Element>
+    </OverflowTooltip>
+  );
+});
+
+export const MenuGroup = createContainer('div')({
+  displayName: 'Menu.Group',
+  modelHook: useMenuGroupModel,
+  elemPropsHook: useMenuGroup,
+  subComponents: {
+    Heading: MenuGroupHeading,
+  },
+})<MenuGroupProps>(({children, title, ...elemProps}, Element) => {
+  return (
+    <Element {...mergeStyles(elemProps)}>
+      {title ? <MenuGroupHeading>{title}</MenuGroupHeading> : null}
```

Comment:
It looks like `MenuGroupHeading` is a sub component of `MenuGroup` so is the `title` prop still valid? 

If the title prop is the only way to add a heading I would either add a data-part attribute so people can style it, or add some css variables for `color` so people can style it. It looks like there's two different APIs here, one where you pass `title` to the `MenuGroup` and one where you can define a `Menu.Group.Heading`.

Reply:
nvm

Reply:
The JSDoc says you can do the following:

```tsx
// default
<Menu.Group title="Group One">
  <Menu.Item>Item One</Menu.Item>
</Menu.Group>

// direct access to the heading
<Menu.Group>
  <Menu.Group.Heading cs={...}>Group One</Menu.Group.Heading>
  <Menu.Item>Item One</Menu.Item>
</Menu.Group>
```

It's kind of a nice API. If you only care about text, you can give a string. If you want access to the underlying component, you have declare the component.

Reply:
I don't know if `title` is the best prop name. It could be `heading` or `label`

Reply:
title is fine, yeah I just saw the jsdoc. I guess we're rolled out "config" APIs for ease of use which I like, it's just not common so yeah I like it! Also `heading` makes sense since it's call `Menu.Group.Heading` but that's nit.

Reply:
I thought about `heading`, but it is the heading text. It doesn't replace the heading component. I think I got the `title` name from React Spectrum's [Section](https://react-spectrum.adobe.com/react-spectrum/Menu.html#sections) which are like groups. `label` could work. I don't want `headingText`. Something short and descriptive. We use `title` for Tooltips which is meant to have text. `title` implies text.